added getplayerchoice and displaynumboard funct

dflotron opened the pull request:
added numerical values to potential board positions so the funt will return an int corresponding the the value. Also added to main a cin to determine X or  0 and send that into my function, you will probably  need that for your functiion as well as knowing the board values

luke-favret committed:
Added most of DisplayBoard func-Doesnt check usrnm

luke-favret committed:
Update README.md

dflotron committed:
Merge pull request #3 from luke-favret/master

displayboard passed by ref

luke-favret committed:
Merge pull request #2 from luke-favret/dflotron-displybrd-fnt

Fixed errors and added displayboard fnt

luke-favret committed:
Merge branch 'master' into luke

luke-favret committed:
Merge branch 'luke' of https://github.com/luke-favret/PE4 into luke

luke-favret committed:
Added PlaceMarker function

dflotron commented:
#include <vector>
#include <iostream>
using namespace std; 
/* Luke Favret and Daisy Flotron
This program is a simple game of tic tac toe */


enum class SquareType { X, O, Empty};


void displayNumBoard() //displays the board with number values in positions to make move selection easier
{
    int arr[3][3]={1,2,3,4,5,6,7,8,9}; 
    for (int i=0; i<3;  i++) {
        for (int j=0; j<3; j++) {
            cout<<" "<<arr[i][j]<<" |"; 
        }
        cout<<endl;
    }

    
}


void DisplayBoard() //arrays passed by ref so correct vals are printed
{
    SquareType board[3][3]; 
    SquareType val; 
    for (int i=0; i<3; i++){
        for (int  j=0;  j<3; j++) {
            val=board[i][j]; 
            if (val==SquareType::X) {
                cout<<" X |"; 
            }
            if (val==SquareType::O) {
                cout<<" 0 |"; 
            }
            if (val==SquareType::Empty) {
                cout<<"   |"; 
            }

        }
        cout<<endl;
    }
}
//Creates board and sets all spots to Empty. Returns pointer to board.
SquareType* CreateBoard(int rows, int columns){ //arrays passed by ref so no need to return it 
    SquareType board[3][3]; //= new SquareType;
    for(int i = 0; i < rows; i ++){ //Set all values in board to Empty
        for(int j = 0; j < columns; j++){
            board[i][j] = SquareType::Empty;
        }
    }
    //return board[3][3];
}
int GetPlayerChoice(char person) //prompts user to determine wheater 
{
    char ans; 
    int move; 
    
    if (person==('X'|'x'))
    {
        DisplayBoard();
        cout<<"Please make a valid sqare selection, input corresponding number value"<<endl;
        displayNumBoard(); 
        cin>>move; 
    }
    if (person=('0'|'o'))
    {
        DisplayBoard();
        cout<<"Please make a valid sqare selection, input corresponding number value"<<endl;
        displayNumBoard();
        cin>>move;
    }
    return move; 
}

int main(){
    int rows = 3;
    int columns = 3;

    SquareType* board = CreateBoard(rows, columns);
    DisplayBoard(); 
    int choice; 
    char person;  
    cout<<"Hello enter player: X or 0"<<endl; //we can put this into a for or while loop to run through the game/turns
    cin>>person; 
    choice=GetPlayerChoice(person);
}

luke-favret commented:
You put the code in the comment box instead of pushing it to your branch then merging your branch into master, so the pull request wouldn't really do anything

luke-favret committed:
Fixed board array. Now correctly declares on heap

dflotron committed:
Merge pull request #5 from luke-favret/luke

Luke

dflotron committed:
Completely changed the structure, added a Board class and now the game is playable and functions as it should

The main needs to be added to in  order to take turns so that the board can be filled out. Try running it (ignore the warnings) and you can see its functional

dflotron committed:
Merge pull request #6 from luke-favret/dflotron-boardclass-1

Completely changed the structure, added a Board class and now the gam…
